SimplyPluralApiCommons.yml

```diff
@@ -39,3 +39,30 @@ components:
         description: Id of the object, this can be a mongo Id but also a plain string (backwards-compatibility).
         schema:
           type: string
+    objectIdOptional:
+      - in: path
+        name: id
+        required: false
+        description: Id of the object, this can be a mongo Id but also a plain string (backwards-compatibility).
```

Comment:
Optional Id of the object. If supplied, this must be a mongodb objectID format. If not supplied, the API will generate the id for you.